File: Challenge 11/c11_easy.py
#make use of date objects
import datetime

#main method is the program
def main():
    #dictionary of weekdays, key is integer value 0 to 6
    weekdays = {0:'Monday', 1:'Tuesday', 2:'Wednesday', 3:'Thursday',
                4:'Friday', 5:'Saturday', 6:'Sunday'}

    #gather the date and split it into a list of integers
    date = [int(x) for x in input("Enter the date (YYYY MM DD): ").split(' ')]

    #create a datetime object
    date = datetime.date(date[0], date[1], date[2])

    #print day of the week based on integer returned from datetime weekday()
    #method (0 - 6)
    print(weekdays[date.weekday()] + ', ' +str(date))

#start the program
main()

File: Challenge 11/c11_easy.pl
#!/usr/bin/perl
#   Daily Programmer Challenge 11 : Easy
#   @author:    Baron Daugherty
#   @date:      2015-07-06
#   @desc:      The program should take three arguments: a day, a month, a year
#               It will then compute the day of the week that date will fall on.
#   @note:      Technically this program does not "compute" the day of the week
#               but rather takes advantage of Perl's included DateTime module.
#               Deal with it.
use warnings;
use strict;
use DateTime;;

#gather the year, month, and day
print("Please enter a date in the the format YYYY MM DD: ");

#create a new DateTime object
my @dates = split(" ", <>);
my $date = DateTime->new(
    year => int($dates[0]),
    month => int($dates[1]),
    day => int($dates[2])
);

#output the result using DT's built in day_name and ymd methods
print($date -> day_name, ", ", $date -> ymd);